packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -1642,16 +1642,92 @@ export interface IBindableBuildImage extends IBuildImage {
   bind(scope: CoreConstruct, project: IProject, options: BuildImageBindOptions): BuildImageConfig;
 }
 
-class ArmBuildImage implements IBuildImage {
+/**
+ * Construction properties of {@link LinuxArmBuildImage}.
+ * Module-private, as the constructor of {@link LinuxArmBuildImage} is private.
+ */
+interface LinuxArmBuildImageProps {
+  readonly imageId: string;
+  readonly imagePullPrincipalType?: ImagePullPrincipalType;
+  readonly secretsManagerCredentials?: secretsmanager.ISecret;
+  readonly repository?: ecr.IRepository;
+}
+
+/**
+ * A CodeBuild image running aarch64 Linux.
+ *
+ * This class has a bunch of public constants that represent the CodeBuild ARM images.
+ *
+ * You can also specify a custom image using the static method:
+ *
+ * - LinuxBuildImage.fromEcrRepository(repo[, tag])
+ *
+ *
+ * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-available.html
+ */
+export class LinuxArmBuildImage implements IBuildImage {
```

Comment:
Can we actually move `LinuxArmBuildImage` (and any types it needs, like `LinuxArmBuildImageProps`) to a separate file? Like we did with the [`LinuxGpuBuildImage` class](https://github.com/aws/aws-cdk/blob/19ead77c6470c0aca4900c4ee200a7c4e66ebbed/packages/%40aws-cdk/aws-codebuild/lib/linux-gpu-build-image.ts#L22-L22)?

This file is already way too big, I'd love to start and chip away at it to make it smaller 🙂.

Reply:
See the open conversation below about a hiccup with this.